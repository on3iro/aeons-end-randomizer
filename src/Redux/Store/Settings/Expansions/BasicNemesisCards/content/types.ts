import { ActionsUnion } from '@martin_hotell/rex-tils'

import * as types from 'aer-types/types'

import { actions } from './actions'
import { LanguageKey } from '../../types'

type LanguageContent = { [id: string]: types.BasicNemesisCard }

export type State = Readonly<
  {
    [key in LanguageKey]: LanguageContent
  }
>

export enum ActionTypes {}

export type Action = ActionsUnion<typeof actions>

export type BasicNemesisCardsContentStateSlice = {
  Settings: {
    Expansions: {
      BasicNemesisCards: {
        content: State
      }
    }
  }
}
