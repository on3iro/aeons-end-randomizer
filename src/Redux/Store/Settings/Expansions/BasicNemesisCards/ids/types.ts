import { ActionsUnion } from '@martin_hotell/rex-tils'

import { actions } from './actions'

export type State = string[]

export enum ActionTypes {}

export type Action = ActionsUnion<typeof actions>

export type BasicNemesisCardIdsStateSlice = {
  Settings: {
    Expansions: {
      BasicNemesisCards: {
        ids: string[]
      }
    }
  }
}
