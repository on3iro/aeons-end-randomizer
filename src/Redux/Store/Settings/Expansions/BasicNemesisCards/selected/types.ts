import { ActionsUnion } from '@martin_hotell/rex-tils'

import { actions } from './actions'

export type State = string[]

export enum ActionTypes {
  TOGGLE_CARD = 'Settings/Expansions/BasicNemesisCards/TOGGLE_CARD',
  SET_TO_DB = 'Settings/Expansions/BasicNemesisCards/SET_TO_DB',
  SET_TO_DB_SUCCESS = 'Settings/Expansions/BasicNemesisCards/SET_TO_DB_SUCCESS',
  SET_TO_DB_FAILURE = 'Settings/Expansions/BasicNemesisCards/SET_TO_DB_FAILURE',
  FETCH_FROM_DB = 'Settings/Expansions/BasicNemesisCards/FETCH_FROM_DB',
  FETCH_FROM_DB_SUCCESS = 'Settings/Expansions/BasicNemesisCards/FETCH_FROM_DB_SUCCESS',
  FETCH_FROM_DB_FAILURE = 'Settings/Expansions/BasicNemesisCards/FETCH_FROM_DB_FAILURE',
}

export type Action = ActionsUnion<typeof actions>

export type SelectedBasicNemesisCardsStateSlice = {
  Settings: {
    Expansions: {
      BasicNemesisCards: {
        selected: string[]
      }
    }
  }
}
