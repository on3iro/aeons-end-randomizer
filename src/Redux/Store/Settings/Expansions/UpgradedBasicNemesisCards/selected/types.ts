import { ActionsUnion } from '@martin_hotell/rex-tils'

import { actions } from './actions'

export type State = string[]

export enum ActionTypes {
  TOGGLE = 'Settings/Expansions/UpgradedBasicNemesisCards/TOGGLE',
  SET_TO_DB = 'Settings/Expansions/UpgradedBasicNemesisCards/SET_TO_DB',
  SET_TO_DB_SUCCESS = 'Settings/Expansions/UpgradedBasicNemesisCards/SET_TO_DB_SUCCESS',
  SET_TO_DB_FAILURE = 'Settings/Expansions/UpgradedBasicNemesisCards/SET_TO_DB_FAILURE',
  FETCH_FROM_DB = 'Settings/Expansions/UpgradedBasicNemesisCards/FETCH_FROM_DB',
  FETCH_FROM_DB_SUCCESS = 'Settings/Expansions/UpgradedBasicNemesisCards/FETCH_FROM_DB_SUCCESS',
  FETCH_FROM_DB_FAILURE = 'Settings/Expansions/UpgradedBasicNemesisCards/FETCH_FROM_DB_FAILURE',
}

export type Action = ActionsUnion<typeof actions>

export type SelectedUpgradedBasicNemesisCardsStateSlice = {
  Settings: {
    Expansions: {
      UpgradedBasicNemesisCards: {
        selected: string[]
      }
    }
  }
}
