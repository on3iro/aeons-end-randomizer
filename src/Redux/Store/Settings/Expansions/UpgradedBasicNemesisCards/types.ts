import { ActionsUnion } from '@martin_hotell/rex-tils'

import * as types from 'types'

import { actions } from './actions'

export type State = Readonly<{
  upgradedBasicNemesisCards: SelectedUpgradedBasicNemesisCards
  upgradedBasicNemesisCardIds: string[]
}>

export enum ActionTypes {
  TOGGLE_CARD = 'Settings/Expansions/UpgradedBasicNemesisCards/TOGGLE_CARD',
  SET_TO_DB = 'Settings/Expansions/UpgradedBasicNemesisCards/SET_TO_DB',
  SET_TO_DB_SUCCESS = 'Settings/Expansions/UpgradedBasicNemesisCards/SET_TO_DB_SUCCESS',
  SET_TO_DB_FAILURE = 'Settings/Expansions/UpgradedBasicNemesisCards/SET_TO_DB_FAILURE',
  FETCH_FROM_DB = 'Settings/Expansions/UpgradedBasicNemesisCards/FETCH_FROM_DB',
  FETCH_FROM_DB_SUCCESS = 'Settings/Expansions/UpgradedBasicNemesisCards/FETCH_FROM_DB_SUCCESS',
  FETCH_FROM_DB_FAILURE = 'Settings/Expansions/UpgradedBasicNemesisCards/FETCH_FROM_DB_FAILURE',
}

export type Action = ActionsUnion<typeof actions>

export type SelectedUpgradedBasicNemesisCards = {
  [key: string]: types.UpgradedBasicNemesisCard & { selected: boolean }
}

export type UpgradedBasicNemesisCardsStateSlice = {
  Settings: {
    Expansions: {
      UpgradedBasicNemesisCards: {
        upgradedBasicNemesisCards: SelectedUpgradedBasicNemesisCards
      }
    }
  }
}

export type UpgradedBasicNemesisCardIdsStateSlice = {
  Settings: {
    Expansions: {
      UpgradedBasicNemesisCards: {
        upgradedBasicNemesisCardIds: string[]
      }
    }
  }
}
