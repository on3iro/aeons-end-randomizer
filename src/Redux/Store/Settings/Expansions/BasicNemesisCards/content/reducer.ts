import { LoopReducer } from 'redux-loop'

import AERData from 'aer-data/src/index'

import { Action, State } from './types'

export const initialState: State = {
  ENG: AERData.normalizedData.ENG.basicNemesisCards,
  FR: AERData.normalizedData.FR.basicNemesisCards,
  PL: AERData.normalizedData.PL.basicNemesisCards,
  DE: AERData.normalizedData.DE.basicNemesisCards,
}

export const Reducer: LoopReducer<State, Action> = (
  state: State = initialState,
  action: Action
) => {
  switch (action.type) {
    default: {
      return state
    }
  }
}
