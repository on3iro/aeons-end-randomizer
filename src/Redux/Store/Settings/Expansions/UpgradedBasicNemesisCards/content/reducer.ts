import { LoopReducer } from 'redux-loop'

import AERData from 'aer-data/src/index'

import { Action, State } from './types'

export const initialState: State = {
  ENG: AERData.normalizedData.ENG.upgradedBasicNemesisCards,
  FR: AERData.normalizedData.FR.upgradedBasicNemesisCards,
  PL: AERData.normalizedData.PL.upgradedBasicNemesisCards,
  DE: AERData.normalizedData.DE.upgradedBasicNemesisCards,
}

export const Reducer: LoopReducer<State, Action> = (
  state: State = initialState,
  action: Action
) => {
  switch (action.type) {
    default: {
      return state
    }
  }
}
