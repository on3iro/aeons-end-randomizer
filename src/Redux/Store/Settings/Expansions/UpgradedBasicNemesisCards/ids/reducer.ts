import { LoopReducer } from 'redux-loop'

import AERData from 'aer-data/src/index'

import { Action, State } from './types'

export const initialState: State =
  AERData.normalizedData.ENG.upgradedBasicNemesisCardIds

export const Reducer: LoopReducer<State, Action> = (
  state: State = initialState,
  action: Action
) => {
  switch (action.type) {
    default: {
      return state
    }
  }
}
