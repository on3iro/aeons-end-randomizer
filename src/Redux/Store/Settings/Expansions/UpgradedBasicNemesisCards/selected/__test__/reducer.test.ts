import { Cmd, getCmd, getModel } from 'redux-loop'

import { State } from '../types'
import { actions } from '../actions'
import {
  getSelectedUpgradedBasicNemesisCardsFromDB,
  setSelectedUpgradedBasicNemesisCardsToDB,
} from '../sideEffects'

import { initialState, Reducer } from '../reducer'

const mockSelectedUpgradedBasicNemesisCardsState: State = initialState

describe('Settings | Expansions | UpgradedBasicNemesisCards | selected | reducer', () => {
  it('should return the initial state', () => {
    // @ts-ignore
    const result = Reducer(undefined, {})

    expect(result).toEqual(initialState)
  })

  it('should handle TOGGLE', () => {
    const selectedUpgradedBasicNemesisCardsToSave = initialState.filter(
      (treasure) => treasure !== 'Wreck'
    )

    const result = Reducer(
      mockSelectedUpgradedBasicNemesisCardsState,
      actions.toggle('Wreck')
    )

    const model = getModel(result)
    const cmd = getCmd(result)

    expect(model).toMatchSnapshot()
    expect(cmd).toEqual(
      Cmd.run(setSelectedUpgradedBasicNemesisCardsToDB, {
        args: [selectedUpgradedBasicNemesisCardsToSave],
        successActionCreator: actions.setToDBSuccessful,
        failActionCreator: actions.setToDBFailed,
      })
    )
  })

  it('should handle FETCH_FROM_DB', () => {
    const result = Reducer(initialState, actions.fetchFromDB())

    const model = getModel(result)
    const cmd = getCmd(result)

    expect(model).toEqual(initialState)

    expect(cmd).toEqual(
      Cmd.run(getSelectedUpgradedBasicNemesisCardsFromDB, {
        successActionCreator: actions.fetchFromDBSuccessful,
        failActionCreator: actions.fetchFromDBFailed,
      })
    )
  })

  it('should handle FETCH_FROM_DB_SUCCESS for defined state', () => {
    const result = Reducer(
      mockSelectedUpgradedBasicNemesisCardsState,
      actions.fetchFromDBSuccessful(['Wreck'])
    )

    expect(getModel(result)).toEqual(['Wreck'])
  })

  it('should handle FETCH_FROM_DB_SUCCESS for undefined state', () => {
    const result = Reducer(
      initialState,
      // @ts-ignore
      actions.fetchFromDBSuccessful(undefined)
    )

    expect(getModel(result)).toEqual(initialState)
  })

  it.todo('should handle FETCH_FROM_DB_FAILURE')
  it.todo('should handle SET_TO_DB')
  it.todo('should handle SET_TO_DB_SUCCESS')
  it.todo('should handle SET_TO_DB_FAILURE')
})
