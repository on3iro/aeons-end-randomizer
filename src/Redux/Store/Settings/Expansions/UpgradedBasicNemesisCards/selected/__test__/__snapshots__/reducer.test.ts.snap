// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings | Expansions | UpgradedBasicNemesisCards | selected | reducer should handle TOGGLE 1`] = `
Array [
  "HissingAcid",
  "BaneCommander",
  "SkyTremor",
  "EngulfingMadness",
  "Separate",
  "NeedleDoom",
  "Marked",
  "Eradicate",
  "FlashOfDecay",
  "Terrify",
  "RayOfDesolation",
  "ThunderousTempest",
  "Sacrifice",
  "Burialskulk",
  "Rupture",
  "FurrowOfDestruction",
  "Mangle",
  "HowlingSlayer",
  "ChaosStrike",
  "LabyrinthWraith-O",
  "RainOfDevastation",
  "HuntTheWeak",
  "Rescind",
  "HarbingerOfMonstrosity-O",
  "MageDevourer-O",
  "SilentDecay",
  "SkyCrush",
  "Scour",
  "Sacrifice-O",
  "ScionOfTheVoid-O",
  "RayOfDesolation-O",
  "BlightedMagic",
  "BlackenedStars",
  "CarriodeRaiders",
  "Incense",
  "SpreadingBrushfire",
  "Blacken",
  "BreachShatter",
  "ChitinKiller",
  "MagmaLordZala",
  "SoulRend",
  "Stifle",
  "DreadIgnition",
  "Explosion",
  "InescapableTorment",
  "NyxOfBurningBones",
  "Ravage",
  "Revive",
  "TheCeaselessObserver",
  "Writhe",
  "BubblingRuin",
  "SweepingBlackness",
  "TwistingFlesh",
  "OmniscientOtherling",
  "ConsumingEarth",
  "Bombard",
  "CoalBeast",
  "Reckoning",
  "Flay",
  "Resurrect",
  "ChainLightning",
  "VoidRevenants",
  "Incense",
]
`;
